sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TypedAggregateExpression.scala

```diff
@@ -34,6 +34,11 @@ object TypedAggregateExpression {
     val bufferEncoder = encoderFor[BUF]
     val bufferSerializer = bufferEncoder.namedExpressions
     val bufferDeserializer = bufferEncoder.deserializer.transform {
+      case UnresolvedAttribute(nameParts) =>
```

Comment:
It seems that the new cleaner implementation does not need my change; it handles unresolved attributes correctly. Thank you!